crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -35,15 +46,59 @@ use std::{
     sync::atomic::{AtomicI64, Ordering},
 };
 
-/// Lightweight unique ID of an entity.
+/// Lightweight identifier of an [entity](crate::entity).
+///
+/// The identifier is implemented using a [generational index]: a combination of an ID and a generation.
+/// This allows fast insertion after data removal in an array while minimizing loss of spatial locality.
+///
+/// [generational index]: https://lucassardois.medium.com/generational-indices-guide-8e3c5f7fd594
+///
+/// # Usage
+///
+/// This data type is returned by iterating a `Query` that has `Entity` as part of its [`WorldQuery`] parameter ([learn more]).
```

Comment:
What's the current stand on the name of the first type paramteter of `Query`? There's no debate that everybody calls the second type parameter “query filter”.

However, I have some concerns about naming the first type parameter `WorldQuery`, because it does not univocally represent it. For example, the query filter is also a `WorldQuery`.

The current name used in the `Query` docs is “Component access”. Could that be appropriate? The sentence would become:

> This data type is returned by iterating a `Query` that has `Entity` as part of its component access type parameter

This example proves that the “component access” name is not correct: in fact, `Entity` is not a component.

Using the name “world access” the sencence would become:

> This data type is returned by iterating a `Query` that has `Entity` as part of its world access type parameter

Reply:
"query fetch" maybe? Component access isn't right because of the widespread use of `Entity` here.

I don't mind "world access" either, but it's a bit less clear IMO and lacks the nice parallels to query filter.